pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2565,10 +2612,73 @@ func (c *Cloud) getTaggedSecurityGroups() (map[string]*ec2.SecurityGroup, error)
 	return m, nil
 }
 
+// Almost identical to updateInstanceSecurityGroupsForLoadBalancer, but it only adds ssg rules to
+// instances' inbound rules.
+func (s *AWSCloud) updateInstanceSharedSecurityGroups(ssgID string, allInstances []*ec2.Instance) error {
```

Comment:
Is it possible to reuse `updateInstanceSecurityGroupsForLoadBalancer`?  Maybe adding a parameter to say "don't remove" if that is the difference?  This function is big enough and tricky enough that I am not wild about copy-pasting it.